tests/unit/lms/views/admin/application_instance_test.py

```diff
@@ -188,6 +189,11 @@ def test_upgrade_duplicate(self, views, lti_registration_service):
 
         assert response.status_code == 400
 
+        # Show that the DB connection has not been permanently broken. This
+        # would cause us to fail completely when trying to present the error.
+        # We are checking we do _not_ raise `PendingRollbackError` here.
+        db_session.query(ApplicationInstance).all()
```

Comment:
You can verify this is doing what it should by removing the rollback above and seeing the test fail.